control/mpc_follower/src/mpc_utils.cpp

```diff
@@ -460,13 +467,13 @@ visualization_msgs::MarkerArray MPCUtils::convertTrajToMarker(
   markers.markers.push_back(marker);
 
   // generate poses
-  visualization_msgs::Marker marker_poses;
+  visualization_msgs::msg::Marker marker_poses;
   marker_poses.header.frame_id = frame_id;
-  marker_poses.header.stamp = ros::Time();
+  marker_poses.header.stamp = rclcpp::Time();
```

Comment:
add `::now()`

Reply:
take proper time stamp as input arg instead of using wallclock time. This way, it works in sim and replay as well